python/vowpalwabbit/pyvw.py

```diff
@@ -1354,3 +1356,656 @@ def get_label(self, label_class=simple_label):
             simple_label
         """
         return label_class(self)
+
+
+class Col:
+    """Col is a convenience class to refer to a column of a dataframe.
+    Its methods can:
+        - check if the column is in a specified dataframe
+        - extract the column from the specified dataframe
+    """
+
+    def __init__(self, colname):
+        """
+        Initialize a Col object
+
+        Parameters
+        ----------
+
+        colname : str
+            The colname that refers to a column
+
+        Raises
+        ------
+
+        TypeError
+            If attribute 'colname' is not a string
+
+        Returns
+        -------
+
+        self : Col
+
+        """
+        if isinstance(colname, str):
+            self.colname = colname
+        else:
+            raise TypeError("'colname' must be a string")
+
+    def col_exist(self, df):
+        """
+        Check if the column 'colname' is in a dataframe 'df'
+
+        Parameters
+        ----------
+
+        df : pandas.DataFrame
+            The dataframe in which to look for the column
+
+        Returns
+        -------
+
+        bool
+            True if the column is in the dataframe, False otherwise.
+
+        """
+        return self.colname in df
+
+    def get_col(self, df):
+        """
+        Extract the column 'colname' from the dataframe 'df'
+
+        Parameters
+        ----------
+
+        df : pandas.DataFrame
+            The dataframe from which to extract the column 'colname'
+
+        Raises
+        ------
+
+        KeyError
+            If the column is not found in the dataframe.
+
+        Returns
+        -------
+
+        out : pandas.Series
+            The column extracted from the dataframe.
+
+        """
+        try:
+            out = df[self.colname].fillna("").apply(str)
+        except KeyError:
+            raise KeyError(
+                "Column '{}' not found in dataframe".format(self.colname)
+            )
+        else:
+            return out
+
+
+def get_col_or_value(x, df):
```

Comment:
Should these functions be "hidden", would the normal Python way be to prefix them like `_get_col_or_value`?

Reply:
We could prefix this function as well as `check_type` and `get_all_cols`. It's convention to say the function is for internal use.
There is no real hidden function/method in Python though, as the user could still import it using `from vowpalwabbit.pyvw import _get_col_or_value`.

To my knowledge the good thing about putting "_" is that the function won't be imported if the user do `from vowpalwabbit.pyvw import *`.